src/foundation/src/PDFsharp/src/PdfSharp/Pdf.Signatures/PdfSignatureHandler.cs

```diff
@@ -110,26 +110,31 @@ private void ComputeSignatureAndRange(object sender, PdfDocumentEventArgs e)
 
         private void AddSignatureComponents(object sender, EventArgs e)
         {
+            if (Options.PageIndex < 0 || Options.PageIndex >= Document.PageCount)
```

Comment:
Don't forget to handle nullity of Options to avoid null ref exception!
for the remaining Options.PageIndex >= Document.PageCount